app/auth/routes.py

```diff
@@ -85,8 +146,7 @@ def user_by_name(id=None,username=None):
             }), 400
     if request.method == 'DELETE':
         try:
-            User.query.filter_by(id=user['id']).delete()
```

Comment:
If the "id" key from "user" does not exist your code will fail. You should use user.get('id',None)